Errors found from scan-build

The following errors were found from clang scan-build static analysis.
Determine if they are real or false positives and fix them.


```
scan-build ./configure $CONFIG_OPTIONS
scan-build make 
scan-build: Using '/usr/lib/llvm-3.8/bin/clang' for static analysis

  CC       lib/tpm2_alg_util.o
lib/tpm2_alg_util.c:350:18: warning: Call to 'malloc' has an allocation size of 0 bytes
        buffer = malloc(*size);
                 ^~~~~~~~~~~~~
1 warning generated.

  CC       tools/tpm2_import.o
tools/tpm2_import.c:169:9: warning: Potential leak of memory pointed to by 'pub_modulus'
        LOG_ERR("RSA_new failed\n");
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
./lib/log.h:57:27: note: expanded from macro 'LOG_ERR'
#define LOG_ERR(fmt, ...) _log(log_level_error, __FILE__, __LINE__, fmt, ##__VA_ARGS__)
                          ^~~~
tools/tpm2_import.c:211:5: warning: Null pointer passed as an argument to a 'nonnull' parameter
    memcpy(to_encrypt_buffer + buffer1_size, buffer2, buffer2_size);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tools/tpm2_import.c:306:33: warning: Result of 'malloc' is converted to a pointer of type 'uint8_t', which is incompatible with sizeof operand type 'TPM2B_PUBLIC'
    uint8_t *marshalled_bytes = malloc(sizeof(ctx.import_key_public));
    ~~~~~~~~~                   ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.

scan-build: 4 bugs found.
```

Background information:
https://clang-analyzer.llvm.org/scan-build.html

I will work on this bug.

--- a/lib/tpm2_alg_util.c
+++ b/lib/tpm2_alg_util.c
@@ -347,6 +347,11 @@ UINT8* tpm2_extract_plain_signature(UINT16 *size, TPMT_SIGNATURE *signature) {
     case TPM2_ALG_HMAC: {
         TPMU_HA *hmac_sig = &(signature->signature.hmac.digest);
         *size = tpm2_alg_util_get_hash_size(signature->signature.hmac.hashAlg);
+        if (*size == 0) {
+            LOG_ERR("Hash algorithm %d has 0 size",
+                signature->signature.hmac.hashAlg);
+            goto nomem;
+        }
         buffer = malloc(*size);
         if (!buffer) {
             goto nomem;

--- a/tools/tpm2_import.c
+++ b/tools/tpm2_import.c
@@ -124,9 +124,10 @@ static int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d) {
 
 static bool encrypt_seed_with_tpm2_rsa_public_key(void) {
     bool rval = false;
-
+    RSA *rsa = NULL;
     TPM2B_PUBLIC pub_key = TPM2B_EMPTY_INIT;
     bool res = files_load_public(ctx.parent_key_public_file, &pub_key);
+
     if (!res) {
         LOG_ERR("Failed loading parent key public file.");
         return false;
@@ -143,7 +144,6 @@ static bool encrypt_seed_with_tpm2_rsa_public_key(void) {
     }
     memcpy(pub_modulus, pub_key_val->buffer, mod_size);
 
-    RSA *rsa = NULL;
     unsigned char encoded[RSA_2K_MODULUS_SIZE_IN_BYTES];
     unsigned char label[10] = { 'D', 'U', 'P', 'L', 'I', 'C', 'A', 'T', 'E', 0 };
     int return_code = RSA_padding_add_PKCS1_OAEP_mgf1(encoded,
@@ -151,24 +151,24 @@ static bool encrypt_seed_with_tpm2_rsa_public_key(void) {
             EVP_sha256(), NULL);
     if (return_code != 1) {
         LOG_ERR("Failed RSA_padding_add_PKCS1_OAEP_mgf1\n");
-        return false;
+        goto error;
     }
     BIGNUM* bne = BN_new();
     if (!bne) {
         LOG_ERR("BN_new for bne failed\n");
-        return false;
+        goto error;
     }
     return_code = BN_set_word(bne, RSA_F4);
     if (return_code != 1) {
         LOG_ERR("BN_set_word failed\n");
         BN_free(bne);
-        return false;
+        goto error;
     }
     rsa = RSA_new();
     if (!rsa) {
         LOG_ERR("RSA_new failed\n");
         BN_free(bne);
-        return false;
+        goto error;
     }
     return_code = RSA_generate_key_ex(rsa, 2048, bne, NULL);
     BN_free(bne);
@@ -208,7 +208,9 @@ static void aes_128_cfb_encrypt_buffers(uint8_t *buffer1, uint16_t buffer1_size,
     //AES encryption
     uint8_t to_encrypt_buffer[max_buffer_size];
     memcpy(to_encrypt_buffer, buffer1, buffer1_size);
-    memcpy(to_encrypt_buffer + buffer1_size, buffer2, buffer2_size);
+    if (buffer2 != NULL) {
+        memcpy(to_encrypt_buffer + buffer1_size, buffer2, buffer2_size);
+    }
 
     uint8_t iv_in[SYM_KEY_SIZE] = { 0 };
     AES_KEY aes;

